drivers/regulator/regulator_mpm54304.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT mps_mpm54304
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mps_mpm54304, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define MPM54304_REG_EN		0x0CU
+#define MPM54304_REG_VENDOR_ID	0x13U
+
+#define MPM54304_buck1_EN_MASK BIT(7)
+#define MPM54304_buck2_EN_MASK BIT(6)
+#define MPM54304_buck3_EN_MASK BIT(5)
+#define MPM54304_buck4_EN_MASK BIT(4)
+
+struct regulator_mpm54304_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec bus;
+	uint8_t enable_mask;
+};
+
+struct regulator_mpm54304_data {
+	struct regulator_common_data common;
+};
+
+static int regulator_mpm54304_enable(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN,
+				      config->enable_mask, 0xFF);
+}
+
+static int regulator_mpm54304_disable(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN,
+				      config->enable_mask, 0x00);
+}
+
+static int regulator_mpm54304_init(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+	uint8_t value;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C bus %s not ready", config->bus.bus->name);
+		return -ENODEV;
+	}
+
+	(void)i2c_reg_read_byte_dt(&config->bus, MPM54304_REG_VENDOR_ID, &value);
+	LOG_DBG("vendor id: 0x%x", value >> 4);
```

Comment:
same, salso, hould the whole block be added under `#idef` log level = dbg?

Reply:
in testing dummy read/write was needed

Reply:
then add a comment about this quirk, please